src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php

```diff
@@ -119,6 +119,18 @@ public function reverseTransform($value)
         }
 
         $formatter = $this->getNumberFormatter();
+        $groupSep = $formatter->getSymbol(\NumberFormatter::GROUPING_SEPARATOR_SYMBOL);
```

Comment:
I duplicate code of `NumberToLocalizedStringTransformer.php` maybe it is better to use it directly ?

`
-        if (!is_string($value)) {
-            throw new TransformationFailedException('Expected a string.');
-        }
-
         if ('' === $value) {
             return;
         }
 
         $formatter = $this->getNumberFormatter();
-        $groupSep = $formatter->getSymbol(\NumberFormatter::GROUPING_SEPARATOR_SYMBOL);
-        $decSep = $formatter->getSymbol(\NumberFormatter::DECIMAL_SEPARATOR_SYMBOL);
         $grouping = $formatter->getAttribute(\NumberFormatter::GROUPING_USED);
-
-        if ('.' !== $decSep && (!$grouping || '.' !== $groupSep)) {
-            $value = str_replace('.', $decSep, $value);
-        }
-
-        if (',' !== $decSep && (!$grouping || ',' !== $groupSep)) {
-            $value = str_replace(',', $decSep, $value);
-        }
-
-        // replace normal spaces so that the formatter can read them
-        $value = $formatter->parse(str_replace(' ', ' ', $value));
-
-        if (intl_is_failure($formatter->getErrorCode())) {
-            throw new TransformationFailedException($formatter->getErrorMessage());
-        }
+        $transformer = new NumberToLocalizedStringTransformer($this->scale, (bool) $grouping);
+        $value = $transformer->reverseTransform($value);
`